seisgo/noise.py

```diff
@@ -42,224 +42,48 @@ def compute_fft(trace,win_len,step,stainv=None,
-                        #if  tr1_len==tr2_len:
-                        trE,trN=utils.correct_orientations(tr1,tr2,orient)
-                        print('Orientation correction for {} is finished'.format(ista))
-                        #else:
-                        #    print('STOP! Error in padding traces. Check start and end times')
-                        #    print(tr1_start)
-                        #    print(tr2_start)
-                        #    print(tr1_end)
-                        #    print(tr2_end)
-                        #    print(tr1.data.shape)
-                        #    print(tr2.data.shape)
-                        #    print(sfile,ista)
-                        #    trE,trN=utils.correct_orientations(tr1,tr2,orient)
-                        #    sys.exit()
-                            
-                        #stE=Stream([trE])
-                        #stN=Stream([trN])
-                        rawdata=[trN,trE]
-
-                    # Skip station if it has less than 2 horizontal channels
-                    elif rotate_flag==1:
-                        print('CONTINUE! Too less chennels for rotation, which requires 2 channels')
-                        return None
-                    # Skip channel rotation (for vertical channel) if no_rotate_flag==1
-                    if no_rotate_flag==1:
-                        for tag in channels:
-                            #print(tag,'was passed directly to fft')
-                            tr,ori,sta_name=trace_info(ds,loc,ista,itag,inv1)
-                            rawdata.append(tr)
-                
-                # Skip orientation correction if correct_orientation is False
-                else:
-                    for itag in all_tags:
-                        if v:print("Assembling raw data for station %s and trace %s" % (ista,itag))
-                        # read waveform data
-                        tr,ori,sta_name=trace_info(ds,loc,ista,itag,inv1)
-                        #tr = ds.waveforms[ista][itag][0]
-                        if len(tr.data)!=0:
-                            rawdata.append(tr)
-
-            raw1['data']=rawdata
-            raw1['inv']=[inv1]
-            raw1['coor']=inv1.get_coordinates(sta_name)
-            raw[ista]=raw1    
-    else:
-        # add liens to extract tags, inv,    
-        raw=dict()
+    elif isinstance(sta,list):
+        sta_list = sta
+    elif isinstance(sta,str):
+        sta_list = [sta]
+
+    # make all work on station list first, if sta is given as a string (station name), then return only the value for that station.
+
+    nsta=len(sta_list)
+    raw_all=[] 
+    if nsta==0:
+        print('no data available')
+        return None
+    sta_list_out=[] #save list of stations with data successfully extracted.
+    for ista in sta_list:
+        # get source station and inventory
+        print('source: {}'.format(ista))
         
-        if do_rotation:
```

Comment:
orientation correction is required for rotation because this step makes horizontal channels the same size. If skip this step, rotation_raw() will have error

Reply:
adding an argument just for another argument is not a good practice in software development. I added a check outside of the function when calling it in do_correlation().